cloud_functions/climateiq_merge_scenario_predictions_cf/main.py

```diff
@@ -0,0 +1,144 @@
+import collections
+import csv
+import re
+
+import flask
+from google.cloud import storage
+import functions_framework
+
+
+# Bucket name, where spatialized prediction outputs are stored.
+INPUT_BUCKET_NAME = "climateiq-spatialized-predictions"
+# Bucket name, where merged prediction outputs are stored.
+OUTPUT_BUCKET_NAME = "climateiq-merged-predictions"
+# File name pattern for the CSVs for each scenario and chunk.
+CHUNK_FILE_NAME_PATTERN = (
+    r"(?P<run_id>\w+)/(?P<prediction_type>\w+)/(?P<model_id>\w+)/"
+    r"(?P<study_area_name>\w+)/(?P<scenario_id>\w+)/(?P<chunk_id>\w+)\.csv"
+)
+
+
+# This only handles merging flood data. Heat data will be divided into completely
+# different chunks.
+@functions_framework.http
+def merge_scenario_predictions(request: flask.Request) -> tuple[str, int]:
```

Comment:
one more case to handle would be if all the H3 indexes are present for all scenarios. if any of the H3 cells are missing, we should throw an error